packages/flutter/lib/src/widgets/form.dart

```diff
@@ -56,8 +65,31 @@ class FormState extends State<Form> {
   int _generation = 0;
   Set<FormFieldState<dynamic>> _fields = new Set<FormFieldState<dynamic>>();
 
-  /// Called when a form field has changed. This will cause all form fields
-  /// to rebuild, useful if form fields have interdependencies.
+  @override
+  void dependenciesChanged() {
+    super.dependenciesChanged();
+    final ModalRoute<dynamic> route = ModalRoute.of(context);
+    if (route != null && route.isCurrent) {
+      // If it's the same ModalRoute, avoid adding our callback twice.
```

Comment:
I'm not sure if you're asking for a test or not understanding the comment. I'll assume that it's both.

The dependenciesChanged() method is called  when the modal route's tree is rebuilt but we don't want to add the same callback twice. Since it's safe to remove it, even it isn't on the route's scopedWillPop callback list, we remove it and then add it back.

I can add a test that verifies that rebuilding the route doesn't cause